src/shared/utils/utils.ts

```diff
@@ -33,34 +33,22 @@ export function formatVideoTime(time: number): string {
  * @returns Fullscreen utils functions
  */
 export function checkSupportFullScreen(): FullScreenSupport {
-	const prefix = getBrowserPrefix()
-
+	const support = [
+		'requestFullscreen,exitFullscreen,fullscreenchange,fullscreenElement',
+		'webkitRequestFullscreen,webkitExitFullscreen,webkitfullscreenchange,webkitFullscreenElement', // older Chrome
+		'mozRequestFullScreen,mozCancelFullScreen,mozfullscreenchange,mozFullScreenElement', // older Firefox
+		'webkitEnterFullscreen,webkitExitFullscreen,webkitfullscreenchange,webkitDisplayingFullscreen' // Safari
+	]
+	const el = document.createElement('video') // on Safari fullscreen is supported only on <video>
+	const items: string[] = (support.find((e) => (el as any)[e.split(',')[0]] instanceof Function) || support[0]).split(',')
```

Comment:
I like the idea of checking whether the method exists on an element, it seems more robust to me!

However, I find the line a bit complex to understand